dhcp-relay/dhcp_kern_xdp.c

```diff
@@ -317,6 +407,16 @@ int xdp_dhcp_relay(struct xdp_md *ctx) {
 	}
 	dhcp = data + vlan_length + dhcp_offset;
 
+	/* Check hops */
+	if (dhcp->hops > 16) {
+		bpf_printk("Max hops exceeded, discarding packet");
+		rc = XDP_ABORTED;
+		goto out;
+	}
```

Comment:
Now we're suddenly doing parsing and sanity checks of the packet in the middle of modifying it. Better to parse and check everything first, *then* start modifying.